Alavian/random drink page fixed api routes

mohzen2 opened the pull request:
Moved API call from frontend to backend, add Cocktail data class for future use

- Removed unnecessary frontend API call and handled it server-side with Ktor
- Added a Cocktail data class (currently unused) to support potential database integration later
- Using raw JSON response for now to simplify API handling

mohzen2 committed:
Added card component for "Random Drink" page navigation and set up routing in App.jsx.

- Created a reusable Card component to display clickable cards with titles and images.
- Added "Random Drink" card on the home page, which links to the /random route.
- Updated App.jsx to include routing for the /random page, rendering the RandomCocktail component for the random drink page.
- Handled the navigation for the "Random Drink" card via react-router-dom's Link component.

mohzen2 committed:
Fixed API routing

# Changes Made
- Created a new backend route `/api/random-drink` in Ktor that proxies requests to CocktailDB
- Updated the RandomCocktail React component to fetch from our backend API instead of directly hitting CocktailDB
- Fixed dependency issues by adding Kotlin serialization plugin and ensuring consistent Ktor version 3.1.2
- Added proper error handling on both frontend and backend

mohzen2 committed:
Fixed API routing

# Changes Made
- Created a new backend route `/api/random-drink` in Ktor that proxies requests to CocktailDB
- Updated the RandomCocktail React component to fetch from our backend API instead of directly hitting CocktailDB
- Fixed dependency issues by adding Kotlin serialization plugin and ensuring consistent Ktor version 3.1.2
- Added proper error handling on both frontend and backend

mohzen2 commented:
Created a separate branch just in case, made a lot of changes this iteration

alexhalcazar commented on src/main/kotlin/com/alexhalcazar/mydrinksapp/Application.kt:
```diff
@@ -7,14 +7,99 @@ import io.ktor.server.http.content.*
+import kotlinx.serialization.json.Json
+
+// Data model for the cocktail response
+@Serializable
```
Please create a model/Cocktail.kt file and move all relevant data and imports to said file.

alexhalcazar commented on src/main/kotlin/com/alexhalcazar/mydrinksapp/Application.kt:
```diff
@@ -7,14 +7,99 @@ import io.ktor.server.http.content.*
+}
+
+// Ktor HTTP client to fetch the cocktail data
+suspend fun fetchRandomCocktail(): String {
```
Since this function fetches data from an external API, let’s keep API calls separate from the data model and repository layers. Please create a service directory (at the same level as model) and move this code into a new file named DrinksApiService.kt.

alexhalcazar reviewed (changes requested):
Major additions to the project! Much appreciated!

mohzen2 committed:
Fixed API routing and removed unnecessary cocktail class

# Changes Made
- Created a new backend route `/api/random-drink` in Ktor that proxies requests to CocktailDB
- Updated the RandomCocktail React component to fetch from our backend API instead of directly hitting CocktailDB
- Added proper error handling on both frontend and backend
Removed the obsolete Cocktail data class from Application.kt
-Created service/DrinksApiService.kt to house all Ktor HTTP‑client calls (e.g. fetchRandomCocktailJson())
-Updated Application.kt to delegate proxy logic to DrinksApiService and slimmed imports
-Removed redundant kotlinx-serialization-core dependency from build.gradle.kts

mohzen2 committed:
Fixed API routing and removed unnecessary cocktail class per request

# Changes Made
- Created a new backend route `/api/random-drink` in Ktor that proxies requests to CocktailDB
- Updated the RandomCocktail React component to fetch from our backend API instead of directly hitting CocktailDB
- Added proper error handling on both frontend and backend
Removed the obsolete Cocktail data class from Application.kt
-Created service/DrinksApiService.kt to house all Ktor HTTP‑client calls (e.g. fetchRandomCocktailJson())
-Updated Application.kt to delegate proxy logic to DrinksApiService and slimmed imports
-Removed redundant kotlinx-serialization-core dependency from build.gradle.kts

mohzen2 commented:
Changes made, second commit was just to remove commented out lines and clean things up
